pkg/reconciler/knativeserving/common/validation.go

```diff
@@ -0,0 +1,260 @@
+/*
+Copyright 2019 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package common
+
+import (
+	"bytes"
+	"fmt"
+	"strings"
+
+	"github.com/rogpeppe/go-internal/semver"
+	v1 "k8s.io/api/core/v1"
+	v1beta1 "k8s.io/api/extensions/v1beta1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	k8sscheme "k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/remotecommand"
+	"knative.dev/pkg/version"
+	opversion "knative.dev/serving-operator/version"
+)
+
+const (
+	// IstioDependency name
+	istioDependency     = "istio"
+	istioNamespace      = "istio-system"
+	istioProductVersion = "productVersion"
+	istioPilotPodLabel  = "istio=pilot"
+	istioPilotContainer = "istio-proxy"
+	// CertManagerDependency name
+	certManagerDependency = "certmanager"
+	certManagerNamespace  = "cert-manager"
+)
+
+type depMeta struct {
+	// dependent
+	name string
+	// in which dependent installed
+	namespace string
+	// deployments installed in the namespace
+	deploy []string
+}
+
+var depMetas = map[string]depMeta{
+	istioDependency: {
+		name:      istioDependency,
+		namespace: istioNamespace,
+		deploy: []string{
```

Comment:
Why do we need this list? we don't really depend on an istio installation even if we depend on istio.

Reply:
We don't need to know the details of Istio deployments. I will update PR soon.